controllers/recipes.js

```diff
@@ -0,0 +1,170 @@
+const Recipe = require("../models/Recipe");
+const Review = require("../models/Review");
+const Comments = require("../models/Comments");
+const User = require("../models/User");
+
+module.exports = {
+  /* CREATE */
+
+  createRecipe: async (request, response) => {
+    const {
+      title,
+      picturePath,
+      servings,
+      steps,
+      ingredients,
+      time,
+      userId,
+      tags,
+    } = request.body;
+
+    const recipe = new Recipe({
+      title,
+      picturePath,
+      servings,
+      steps,
+      ingredients,
+      time,
+      userId,
+      tags,
+    });
+
+    const user = request.user;
+    if (!user) {
+      return responseponse.status(401).json({error: "operation not permitted"});
+    }
+
+    recipe.userId = user._id;
+
+    const createdRecipe = await recipe.save();
+    user.recipes = user.recipes.concat(createdRecipe._id);
+    response.status(201).json(createdRecipe);
+  },
+
+  createReview: async (request, response) => {
+    const recipe = await Recipe.findById(request.params.id);
+    const user = request.user;
+
+    const {timesMade, rating, picturePath} = request.body;
+
+    const review = new Review({
+      recipeId: recipe._id,
+      reviewer: user._id,
+      timesMade,
+      rating,
+      picturePath,
+    });
+    const createdReview = await review.save();
+    user.reviews = user.reviews.concat(createdReview._id);
+    await user.save();
+    recipe.review = recipe.review.concat(createdReview._id);
+    await recipe.save();
+
+    response.status(201).json(createdReview);
+  },
+  createComment: async (request, response) => {
+    const recipe = await Recipe.findById(request.params.id);
+    const user = request.user;
+    const {parentId, text} = request.body;
+
+    const comment = new Comments({
+      recipeId: recipe._id,
+      userId: user._id,
+      parentId,
+      text,
+    });
+
+    const createdComment = await comment.save();
+
+    recipe.comments = recipe.comments.concat(createdComment._id);
+    await recipe.save();
+    response.status(201).json(createdComment);
+  },
+
+  /* READ */
+
+  getFeedRecipes: async (request, response) => {
```

Comment:
What's the difference between getFeedRecipes and getRecipe?

Reply:
getRecipe - get only one specfic recipe
and get FeedRecipe - is all the recipes from the database

Do you think getOneRecipe is more clear?